### ISSUE TITLE
How to add required Authorization/Security to the OpenAPI spec

### ISSUE THREAD

**Issue by BradNut** [paolostyle/hono-zod-openapi#3]
It would be nice to be able to setup authorization per route which can be consumed by an OpenAPI client as such.

For example, following the `zod-openapi` middleware [how-to-setup-authorization](https://github.com/honojs/middleware/tree/main/packages/zod-openapi#how-to-setup-authorization).

Adding a header for authorization can partly satisfy the requirement as it does add it to the document but it doesn't seem to enforce when consumed through something like `@scalar/hono-api-reference`.

When I add something like:
```
request: {
    header: z.object({
        authorization: z.string(),
    }),
    json: updateProfileDto,
},
```

It does produce:
```
"parameters": [{
    "in": "header",
    "name": "authorization",
    "schema": {
        "type": "string"
    },
    "required": true
}],
"requestBody": {
    "content": {
```

But when read via something like scalar I only see it as a required request item but not a required authentication type. I know this probably has not been implemented but here is an example of what it looks like using `zod-openapi`:
```
app.openAPIRegistry.registerComponent("securitySchemes", "Bearer", {
  type: "http",
  scheme: "bearer",
});

//...
export const remove = createRoute({
  path: "/tasks/{id}",
  method: "delete",
  security: [
    {
      Bearer: [],
    },
  ],
```

With output in Scalar: 
![OpenAPI_Auth_Type](https://github.com/user-attachments/assets/aff9d014-81ae-4f5d-af39-d12a503c7e91)

In your library all I can do is something like:
![Required_In_Definition](https://github.com/user-attachments/assets/fada8105-8a3a-432b-98d5-d560b2dc09f5)

Which doesn't result in anything in the request:
![Not_Required_In_Test](https://github.com/user-attachments/assets/7968f093-0fd3-44fe-a862-9c5a6111cdf1)

**Comment by paolostyle** [comment]
It is possible to do this, but it's not really documented here as there's no high-level way to do this. Equivalent in `hono-zod-openapi` would be something like this:

```
    const app = new Hono<{ Bindings: { env: 'test' } }>().basePath('/api').get(
      '/example',
      openApi({
        responses: {
          200: z.object({}),
        },
        security: [{ bearerAuth: [] }],
      }),
      async (c) => {
        return c.json({}, 200);
      },
    );

    createOpenApiDocument(app, {
      info: {
        title: 'Some API',
        version: '0.0.1',
      },
      components: {
        securitySchemes: {
          bearer: {
            type: 'http',
            scheme: 'bearer',
          },
        },
      },
      security: [
        {
          bearerAuth: [],
        },
      ],
    });
```

It's pretty much one to one with the OpenAPI guide from Swagger: https://swagger.io/docs/specification/v3_0/authentication/bearer-authentication/.

I'm open to providing a higher level API to manage this, although I don't have a good idea at the moment how it could look like.

**Comment by BradNut** [comment]
Ah I see now @paolostyle. Yes I am able to replicate using your example above and adding to my API definition like:
```
tags,
security: [{ bearerAuth: [] }],
request: {
  json: mySchema,
},
responses: {
},
```

If you figure out something less tedious then that's great. Otherwise I can use this. Thank you!

**Comment by paolostyle** [comment]
Actually it seems like I missed an important part; the `security` key in `createOpenApiDocument` is only necessary if you want to use it globally (I adjusted the example above). If all your endpoints will require bearer authentication you can use that instead of setting `security: [{ bearerAuth: [] }]` for each route. If, let's say, majority, but not all of your endpoints will use it, unfortunately you need to add it to each endpoint.

I just released v0.3.0 which adds a type required to do this, so after updating you'll be able to create your own wrapper with common properties. E.g. something like this:

```ts
const taggedAuthRoute = <T extends HonoOpenApiRequestSchemas>(
  tag: string,
  doc: HonoOpenApiOperation<T>,
) => {
  return defineOpenApiOperation({
    ...doc,
    tags: [tag],
    security: [{ apiKey: [] }],
  });
};

// then use like this
openApi(taggedAuthRoute('MyTag', { responses: /* ... */} ))
```

**Comment by BradNut** [comment]
@paolostyle , yes you are correct. Thank you! I updated and am able to use it like you have above.
